kymatio/scattering2d/frontend/numpy_frontend.py

```diff
@@ -0,0 +1,45 @@
+from ...frontend.numpy_frontend import ScatteringNumPy
+from kymatio.scattering2d.core.scattering2d import scattering2d
+from .base_frontend import ScatteringBase2D
+from ..utils import compute_padding
+from ..filter_bank import filter_bank
+import numpy as np
+
+class ScatteringNumPy2D(ScatteringNumPy, ScatteringBase2D):
+    def __init__(self, J, shape, L=8, max_order=2, pre_pad=False, backend='numpy'):
+        ScatteringNumPy.__init__(self)
+        ScatteringBase2D.__init__(**locals())
+        ScatteringBase2D._instantiate_backend(self, 'kymatio.scattering2d.backend.')
+        ScatteringBase2D.build(self)
+        ScatteringBase2D.create_filters(self)
+    
+    def scattering(self, input):
+        if not type(input) is np.ndarray:
+            raise TypeError('The input should be a NumPy array.')
+
+        if len(input.shape) < 2:
+            raise RuntimeError('Input array must have at least two dimensions.')
+
+        #if not input.is_contiguous():
+        #    raise RuntimeError('Tensor must be contiguous.')
+
+        if (input.shape[-1] != self.N or input.shape[-2] != self.M) and not self.pre_pad:
+            raise RuntimeError('NumPy array must be of spatial size (%i,%i).' % (self.M, self.N))
+
+        if (input.shape[-1] != self.N_padded or input.shape[-2] != self.M_padded) and self.pre_pad:
+            raise RuntimeError('Padded array must be of spatial size (%i,%i).' % (self.M_padded, self.N_padded))
+
+        batch_shape = input.shape[:-2]
+        signal_shape = input.shape[-2:]
+
+        input = input.reshape((-1,) + signal_shape)
```

Comment:
Perhaps there should be a function in backend where the job of the function is to do this stuff. Honestly, the batch_shape/signal_shape stuff could be a utility function, as the backends uses this code often.

Reply:
Good idea.

Reply:
yes

Reply:
So I think we just leave this for now, but in the future, it might be nice to have two functions: `x, batch_shape = compress_batch(x, dim)`, where `dim` corresponds to how many dimensions to “keep” (here, two), and `x = restore_batch(x, batch_shape)`. This way, we can use the functions across dimensions (and frontends) by changing the `dim` parameter.

Reply:
Made this into an issue. Resolving.